lit_gpt/lora.py

```diff
@@ -589,15 +582,15 @@ def __init__(self, config: Config) -> None:
             config.n_embd,
             config.intermediate_size,
             bias=config.bias,
-            r=config.r,
+            r=(config.r if config.to_mlp else 0),
```

Comment:
In `lora.py` (and maybe other files using LoRA), the default value that will set `config.to_mlp` is `False` (https://github.com/Lightning-AI/lit-gpt/blob/main/finetune/lora.py#L46). Do we want https://github.com/Lightning-AI/lit-gpt/blob/main/finetune/lora.py#L46 to be False by default?

Reply:
The original LoRA paper only applies it to Q, and V. Later papers suggest applying it to other layers so support was added by always converting the layer to LoRA but controling whether it's enabled via `r`.

So `to_mlp` should be kept as `False` to be consistent with the original LoRA paper.

This PR fixes that the MLP linear layers weren't getting converted as the rest of the linears.